### PR TITLE
fix bug with endcard not showing

### PR DESCRIPTION
**Pull request by EdmundsBankai**
# 📌 - Related Issue
- [KVID-557](https://kargo1.atlassian.net/browse/KVID-557)

# 🏹 - Proposed Changes
- Refactor/Bug Fix - move `this.setEndCardBackground()` to ad start event because the endcard was not properly being captured and shown

# :dependabot:  Dependencies
- None

# 🚨 - Schema Updates
- None

# 🚧 - Impacted Areas
- `this.setEndCardBackground()` moved to ad start event

# 🎯 - Checklist
- [x] Tested locally
- [ ] Unit Test
- [ ] Add milestone, labels, reviewers

# 🎬 - Demo
- [Demo](https://drive.google.com/file/d/1NOkibqWRyjEFCcxtbaTqX9JyVOdQKA4r/view?usp=sharing)

### CODE REVIEW

**Review by sambua — CHANGES_REQUESTED** [review]
Hi @EdmundsBankai code is looks great. But it require changes.

1. `ads:start` event is too late event, it can be trigger later when we could require this.get("ad") before to use.
2. Actually `ad.data.mediaUrl` URL which you're using is available on `ads:loaded` event as well. 
3. Reason why it works as expected on your new changes is that video is already loaded on `ads:start` and cached by Chrome so you can easily run `video.play()` function.
4. Will be better to replace setTimeout part inside `generateEndCardImage` with Promise as code as per below. I've tested locally and seems works as expected from my end.
```
                    this._mediaUrl = ad?.data?.mediaUrl;
                    fetch(this._mediaUrl)
                        .then(response => response.blob())
                        .then(blob => {
                          this._video.src = URL.createObjectURL(blob);
                          return this._video.play();
                        })
                        .then(_ => {
                          // draw canvas logic here 
                          // and pause video at the end...
                        })
                        .catch(e => {})
``` 
Thanks in advance!

**Comment by EdmundsBankai** [comment]
@sambua New PR with requested changes here. Closing this one
https://github.com/betajs/betajs-media-components/pull/512

File: src/dynamics/ads_player/ads_player.js [hunk 1]
```diff
@@ -381,34 +364,61 @@ Scoped.define("module:Ads.Dynamics.Player", [
...
+                    const width = this.getAdWidth();
+                    const height = this.getAdHeight();
+                    if (width && height) {
+                        if (this.adsManager && typeof this.adsManager.resize === "function") {
+                            this.adsManager.resize(
+                                width,
+                                height,
+                                google.ima.ViewMode.NORMAL
+                            );
+                        }
```
Comment (ntshcalleia) [inline src/dynamics/ads_player/ads_player.js:384]: Hi @EdmundsBankai, this part isn't related to setting end card background, it's something we need to do whenever the player is resized to ensure the IMA SDK has the correct sizing, so I don't think it should be wrapped inside a function called `setEndCardBackground`.

Comment (EdmundsBankai) [reply src/dynamics/ads_player/ads_player.js:384]: yes, i didn't notice that, will update this

**Pull request closed without merging**